oak-core/src/main/java/org/apache/jackrabbit/oak/query/ast/FullTextSearchImpl.java

```diff
@@ -137,27 +138,29 @@ public FullTextExpression getFullTextConstraint(SelectorImpl s) {
             }
             String p2 = normalizePropertyName(p);
             String rawText = getRawText(v);
-            FullTextExpression e = FullTextParser.parse(p2, rawText);
+            InferenceQuery inferenceQuery = new InferenceQuery(rawText);
+            String queryText = inferenceQuery.getQueryText();
```

Comment:
Could you mention in a comment why exactly is the InferenceQuery object needed here ? I am assuming some parsing required for rawtext->queryText for some specific case ?